### PR TITLE
Fix logic of wayland if statement

### PR DESCRIPTION
**Pull request by dylantjb**

### CODE REVIEW

File: zsh-system-clipboard.zsh [hunk 1]
```diff
@@ -41,7 +41,7 @@ case "$OSTYPE" {
 		fi
 		;;
 	linux*|freebsd*)
-		if _zsh_system_clipboard_command_exists wl-copy && [[ -z "$DISPLAY" ]] && [[ -z "$ZSH_SYSTEM_CLIPBOARD_USE_WL_CLIPBOARD" ]]; then
+		if _zsh_system_clipboard_command_exists wl-copy && [[ -z "$DISPLAY" ]] || [[ "$ZSH_SYSTEM_CLIPBOARD_USE_WL_CLIPBOARD" ]]; then
```
Comment (doronbehar) [inline zsh-system-clipboard.zsh]: What is the difference? Maybe I'm missing some knowledge in ZSH booleans..

Comment (dylantjb) [reply zsh-system-clipboard.zsh]: The -z option checks if the variable is null.
My edit makes sure if the variable is used, wayland is used regardless.

Comment (doronbehar) [reply zsh-system-clipboard.zsh]: Could you explain what do you mean by "wayland is used regardless"? What is your scenario? Do you have `$DISPLAY` set in your environment although you use wayland?

Comment (dylantjb) [reply zsh-system-clipboard.zsh]: I mean if ZSH_SYSTEM_CLIPBOARD_USE_WL_CLIPBOARD is set to anything, then wl-clip is used. At least in my case, Sway sets DISPLAY regardless of me exporting it when launching from a tty, so I edited this to make it work.

Comment (dylantjb) [reply zsh-system-clipboard.zsh]: This is most likely down to xorg-xwayland so if I unset it, then I cannot open any X applications.

Comment (b10n) [reply zsh-system-clipboard.zsh]: I also noticed that `ZSH_SYSTEM_CLIPBOARD_USE_WL_CLIPBOARD` is ignored. This PR fixes that for me.

Using the `-v` switch in `[[ ... ]]` makes it consistent with [this description](https://github.com/kutsan/zsh-system-clipboard/blob/f9349b1e51a5e9852ee146256790094b70a18964/zsh-system-clipboard.zsh#L103), i.e. setting the variable to the empty string enables it too.
```suggestion
		if _zsh_system_clipboard_command_exists wl-copy && [[ -z "$DISPLAY" ]] || [[ -v ZSH_SYSTEM_CLIPBOARD_USE_WL_CLIPBOARD ]]; then
```

**Comment by doronbehar** [comment]
OK I don't understand really what's the issue, but I trust you @b10n and @dylantjb that you use this variable and that this change is needed. @dylantjb please squash the commits and I'll approve the PR.

**Comment by idest** [comment]
Without this change if `wl-copy` is installed and `$DISPLAY` is set, the plugin always skips the `_zsh_system_clipboard_set` and `_zsh_system_clipboard_get` assignments, regardless of wheter the environment variable `ZSH_SYSTEM_CLIPBOARD_USE_WL_CLIPBOARD` is set. With this change the plugin behaves exactly as described in the README.

**Comment by doronbehar** [comment]
Thanks for the reminder this is still open, I haven't received a notification for the force push.

**Comment by pbrilius** [comment]
Good **insight**.

**Pull request merged**